static/js/upload.js

```diff
@@ -13,187 +17,217 @@ exports.feature_check = function (upload_button) {
+    if (page_params.max_file_upload_size === 0) {
+        exports.show_error_message(config, i18n.t('File and image uploads have been disabled for this organization.'));
+        return;
+    }
+    exports.get_item("send_button", config).attr("disabled", "");
+    exports.get_item("send_status", config).addClass("alert-info").removeClass("alert-error").show();
+    exports.get_item("send_status_message", config).html($("<p>").text(i18n.t("Uploading…")));
+    exports.get_item("send_status_close_button", config).one('click', function () {
+        uppy.cancelAll();
+        setTimeout(function () {
+            exports.hide_upload_status(config);
+        }, 500);
+    });
+
+    for (const file of files) {
+        try {
+            compose_ui.insert_syntax_and_focus("[Uploading " + file.name + "…]()", exports.get_item("textarea", config));
+            compose_ui.autosize_textarea();
+            uppy.addFile({
+                source: exports.get_item("source", config),
+                name: file.name,
+                type: file.type,
+                data: file,
+            });
+        } catch (error) {
+            // Errors are handled by info-visible and upload-error event callbacks.
         }
+    }
+};
 
-        switch (error_code) {
-        case 'BrowserNotSupported':
-            msg = i18n.t("File upload is not yet available for your browser.");
-            break;
-        case 'TooManyFiles':
-            msg = i18n.t("Unable to upload that many files at once.");
-            break;
-        case 'FileTooLarge':
-            if (page_params.max_file_upload_size > 0) {
-                // sanitization not needed as the file name is not potentially parsed as HTML, etc.
-                var context = {
-                    file_name: file.name,
-                    file_size: page_params.max_file_upload_size,
-                };
-                msg = i18n.t('"__file_name__" was too large; the maximum file size is __file_size__MB.',
-                             context);
-            } else {
-                // If uploading files has been disabled.
-                msg = i18n.t('File and image uploads have been disabled for this organization.');
-            }
-            break;
-        case 413: // HTTP status "Request Entity Too Large"
-            msg = i18n.t("Sorry, the file was too large.");
-            break;
-        case 400: {
-            var server_message = server_response && server_response.msg;
-            msg = server_message || i18n.t("An unknown error occurred.");
-            break;
+exports.setup_upload = function (config) {
+    const uppy = Uppy({
+        debug: true,
+        autoProceed: true,
+        restrictions: {
+            maxFileSize: page_params.max_file_upload_size * 1024 * 1024,
+        },
+        locale: {
+            strings: {
+                exceedsSize: i18n.t('This file exceeds maximum allowed size of'),
+                failedToUpload: i18n.t('Failed to upload %{file}'),
```

Comment:
Convention for marking variables in i18n strings is a bit diffrent than our approach. This should not be an issue for translators right? We might have to update the docs.

Reply:
I’m not sure it matters, but you may be able to retain compatibility with the existing approach with `i18n.t('Failed to upload __file__', { file: '%{file}' })`.

Reply:
Actually, the existing approach is probably strictly worse since Transifex doesn’t automatically recognize `__underscores__` as delimiting variables… why did we decide to use those?

Reply:
@timabbott might have better context on this.

Reply:
@andersk I don't recall the history of the `__underscores__` choice for variables; I think we just failed to check that aspect of how Transifex handles things and used the library's recommendation.  Would be fine to migrate to another syntax in a sweep.  Seems like a migration we should just do in an independent commit, so I'd just do the interpolation the way the library expects for this, but good catch on us wanting to do a sweet here.